src/main/java/dev/morling/onebrc/CalculateAverage_gauravdeshmukh.java

```diff
@@ -0,0 +1,304 @@
+/*
+ *  Copyright 2023 The original authors
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+package dev.morling.onebrc;
+
+import java.io.File;
+import java.io.RandomAccessFile;
+import java.nio.ByteBuffer;
+import java.nio.MappedByteBuffer;
+import java.nio.channels.FileChannel;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.TreeMap;
+import java.util.concurrent.Callable;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+
+public class CalculateAverage_gauravdeshmukh {
+
+    private static final String FILE = "./measurements.txt";
+    private static final byte NEGATIVE_SIGN_BYTE = 0x2D;
+    private static final byte DOT_BYTE = 0x2E;
+    private static final int SEARCH_SPACE_BUFFER_SIZE = 140;
+
+    private static final long SEMI_COLON_MASK = 0x3B3B3B3B3B3B3B3BL;
+    private static final long EOL_MASK = 0x0A0A0A0A0A0A0A0AL;
+
+    private static class ByteString {
+        final private String stringBytes;
+        final private int staticHashCode;
+
+        public ByteString(byte[] bytes) {
+            this.stringBytes = new String(bytes, StandardCharsets.UTF_8);
+            this.staticHashCode = this.stringBytes.hashCode();
+        }
+
+        public byte[] getBytes() {
+            return stringBytes.getBytes(StandardCharsets.UTF_8);
+        }
+
+        @Override
+        public boolean equals(Object bs) {
+            return bs.hashCode() == this.staticHashCode;
```

Comment:
How does that handle hash collisions?

Reply:
It doesn't. It relies on low probability of collision given the max size of string being 100 bytes.

Reply:
Updated PR to remove that assumption and to use default string equals method. There are no changes in the execution time on local machine.

Reply:
@gunnarmorling  can you please check updated PR?